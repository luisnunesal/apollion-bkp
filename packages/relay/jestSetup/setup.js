const { JSDOM } = require('jsdom');
const server = require('./testServer');

module.exports = async () => {
  const httpServer = await server.listen(5000, () => {
    console.log(`\n test server running at: \n${JSON.stringify(httpServer.address(), null, 2)}`);
  });

  Object.assign(global, { httpServer });
};
