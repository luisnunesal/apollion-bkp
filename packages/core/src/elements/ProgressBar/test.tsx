import 'jest-styled-components';

import React from 'react';

import { ProgressBar } from '.';

import { render, screen } from '../../__tests__/setup';

describe('ProgressBar', () => {
  // ---- CONTENT ---- //
  it('Should render successfully', () => {
    render(<ProgressBar progress={10} />);
    expect(screen).toMatchSnapshot();
  });

  // ---- COLORS ---- //
  it('Snapshot should match color="warning"', () => {
    render(<ProgressBar progress={10} color="warning" />);
    expect(screen).toMatchSnapshot();
  });

  // ---- PROGRESS LABEL ---- //
  it('Progress label should be visible ', () => {
    render(<ProgressBar color="warning" progress={10} showProgressLabel />);
    expect(screen.getByText('10%')).toBeDefined();
  });
});
