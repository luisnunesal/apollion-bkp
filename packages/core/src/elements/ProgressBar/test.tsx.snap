// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render successfully 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  margin: 0;
  width: 100%;
}

.c0 .sc-AxirZ {
  color: #144658;
}

.c1 {
  width: 100%;
  height: 0.75rem;
  border-radius: 0.75rem;
  margin-bottom: 0.5rem;
  border-radius: 0;
  background: #EBF7FC;
  overflow: hidden;
}

.c2 {
  width: 100%;
  height: 100%;
  background: #32AFDC;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
}

<div
  class="sc-AxjAm c0"
  color="primary"
>
  <div
    class="c1"
    color="primary"
  >
    <div
      class="c2"
      color="primary"
      style="transform: translateX(-90%);"
    />
  </div>
</div>
`;

exports[`Button Snapshot should match color="warning" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  margin: 0;
  width: 100%;
}

.c0 .sc-AxirZ {
  color: #664B00;
}

.c1 {
  width: 100%;
  height: 0.75rem;
  border-radius: 0.75rem;
  margin-bottom: 0.5rem;
  border-radius: 0;
  background: #FFF8E6;
  overflow: hidden;
}

.c2 {
  width: 100%;
  height: 100%;
  background: #FFBB00;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
}

<div
  class="sc-AxjAm c0"
  color="warning"
>
  <div
    class="c1"
    color="warning"
  >
    <div
      class="c2"
      color="warning"
      style="transform: translateX(-90%);"
    />
  </div>
</div>
`;
